parquet-hadoop/src/main/java/org/apache/parquet/crypto/InternalFileDecryptor.java

```diff
@@ -230,8 +232,8 @@ public InternalColumnDecryptionSetup setColumnCryptoMetadata(ColumnPath path, bo
           // No explicit column key given via API. Retrieve via key metadata.
           try {
             columnKeyBytes = keyRetriever.getKey(keyMetadata);
-          } 
-          catch (KeyAccessDeniedException e) {
+          }
+          catch (KeyAccessDeniedException | AccessControlException | AccessDeniedException e) {
```

Comment:
keyRetriever is your code. You can catch AccessControlException | AccessDeniedException e there, and throw new KeyAccessDeniedException(e)

Reply:
by your code, I mean its always a custom implementation of the DecryptionKeyRetriever interface. - which can catch AccessControlException | AccessDeniedException e there, and throw new KeyAccessDeniedException(e).